colcon_gradle/task/gradle/build.py

```diff
@@ -56,14 +58,25 @@ def add_arguments(self, *, parser):  # noqa: D102
             help='Run a specific task instead of the default task')
 
     async def build(
-        self, *, additional_hooks=None, skip_hook_creation=False
+        self, *, additional_hooks=[], skip_hook_creation=False
     ):  # noqa: D102
         pkg = self.context.pkg
         args = self.context.args
 
         logger.info(
             "Building Gradle package in '{args.path}'".format_map(locals()))
 
+        # add jars and classes to CLASSPATH with wildcards
+        # https://docs.oracle.com/javase/8/docs/technotes/tools/windows/classpath.html#A1100762
+        additional_hooks += create_environment_hook(
+            'classpath_jars', Path(args.install_base), pkg.name,
+            'CLASSPATH', os.path.join('share', pkg.name, 'java', '*'),
+            mode='prepend')
+        additional_hooks += create_environment_hook(
```

Comment:
do we also need the directory?
I think that the wildcard is enough

Reply:
Reading the the [docs](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/classpath.html#A1100762), I'm lead to believe we need both the path and the wildcard to account for class files AND jars:

> A class path entry that contains an asterisk (*) does not match class files. To match both classes and JAR files in a single directory mydir, use either mydir:mydir/* or mydir/*:mydir. The order chosen determines whether the classes and resources in mydir are loaded before JAR files in mydir or vice versa.